### PR TITLE
tag pill refactor

### PR DESCRIPTION
**Pull request by stuartromanek**
- move Close button into component
- knobs
- rename component in line with standards
- nudge

### CODE REVIEW

File: src/components/tag/AposTag.vue [hunk 1]
```diff
@@ -1,38 +1,46 @@
...
   justify-content: center;
+}
 
-  .material-design-icon {
-    display: flex;
-    margin-right: 5px;
+.apos-tag__close-icon {
+  display: flex;
+  position: relative;
+  top: 1px;
```
Comment (abea) [inline src/components/tag/AposTag.vue]: This seems to work with _either_ `display: flex` _or_ the relative `top: 1px` business. With both it's looking off center. I'd go with flex or `inline-flex`, probably.

Screen Shot 2020-03-24 at 9 48 02 AM

File: src/components/tag/AposTag.vue [hunk 2]
```diff
@@ -1,38 +1,46 @@
 <template>
   <div :class="`apos-tag ${modifier}`" :disabled="disabled">
```
Comment (abea) [inline src/components/tag/AposTag.vue]: The `disabled` attribute doesn't do anything anymore.

Comment (abea) [reply src/components/tag/AposTag.vue]: Can't tell if it ever did. Maybe this used to be a button... Should this be a button? I think that might make sense. It does a thing when you click it. The alternative would be to make only the `x` a button.

Comment (abea) [reply src/components/tag/AposTag.vue]: So then we should be emitting an event when it's clicked as well.

File: src/components/tag/AposTag.stories.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+export const tags = () => ({
+  components: { AposTag },
+  props: {
+    label: {
+      default: text('Label', 'Neighborhood')
+    }
+  },
+  template: `
+    <AposTag
+      :label="label"
```
Comment (abea) [inline src/components/tag/AposTag.stories.js]: Will it make sense to be passing a prop for the close icon? Thinking out loud here.

Comment (abea) [reply src/components/tag/AposTag.stories.js]: The preemptive (before we're hitting 80 chars) line-breaks make this a bit harder to parse quickly, imo.

File: src/components/tag/AposTag.vue (hunk 1 above)
Comment (abea) [reply src/components/tag/AposTag.vue]: Nevermind. You're doing relative/1px on notifications and that one doesn't respond to `flex`, so let's keep it consistent using the relative positioning.

File: src/components/tag/AposTag.vue (hunk 2 above)
Comment (stuartromanek) [reply src/components/tag/AposTag.vue]: going to make the x a button, clicking the whole thing to remove feels bad

File: src/components/tag/AposTag.stories.js (hunk 3 above)
Comment (stuartromanek) [reply src/components/tag/AposTag.stories.js]: make sense in that we should leave it open to receive an icon?

Comment (abea) [reply src/components/tag/AposTag.stories.js]: Not an icon, but for whether it's closable or not. Maybe some will close on their own always?

File: src/components/tag/AposTag.vue [hunk 4]
```diff
@@ -10,18 +12,19 @@ import Close from 'vue-material-design-icons/Close.vue';
 export default {
   props: {
     label: String,
-    modifier: {
-      default: '',
-      type: String
-    }
+    slug: String
```
Comment (abea) [inline src/components/tag/AposTag.vue]: This doesn't seem to be used.

File: src/components/tag/AposTag.stories.js (hunk 3 above)
Comment (stuartromanek) [reply src/components/tag/AposTag.stories.js]: close on their own? these are tag UI

File: src/components/tag/AposTag.vue [hunk 5]
```diff
@@ -1,6 +1,8 @@
 <template>
-  <div :class="`apos-tag ${modifier}`" :disabled="disabled">
-    <Close class="apos-tag__close-icon" title="Remove Tag" :size="10" />
+  <div :class="`apos-tag`">
```
Comment (abea) [inline src/components/tag/AposTag.vue]: we can go with the regular syntax now.

File: src/components/tag/AposTag.stories.js [hunk 6]
```diff
@@ -12,11 +12,10 @@ export const tags = () => ({
   props: {
     label: {
       default: text('Label', 'Neighborhood')
+    },
+    slug: {
```
Comment (abea) [inline src/components/tag/AposTag.stories.js]: this doesn't look to be used.

File: src/components/tag/AposTag.vue [hunk 7]
```diff
@@ -1,6 +1,8 @@
...
+    <button class="apos-tag-button">
+      <Close class="apos-tag__close-icon" @click="click" title="Remove Tag" :size="10" />
```
Comment (abea) [inline src/components/tag/AposTag.vue]: The `@click` jawn should be on the button.

**Comment by stuartromanek** [comment]
- emit slug in event to inform some action from parent
- cleanup

not 100% it will be called `slug` or whatever, but just as easy to change

**Pull request merged**